src/core/thread/mle.cpp

```diff
@@ -3874,7 +3946,270 @@ void Mle::HandleLinkProbe(const Message &aMessage, const Ip6::MessageInfo &aMess
+    {
+        error = Get<MleRouter>().HandleLinkAccept(aMessage, aMessageInfo, aKeySequence, aNeighbor, true);
+    }
+    else
+#endif
+    {
+#if OPENTHREAD_CONFIG_MAC_SSED_TO_SSED_LINK_ENABLE
+        error = HandleLinkAccept(aMessage, aMessageInfo, true);
+#endif
+    }
+
+exit:
+    LogProcessError(kTypeLinkAcceptAndRequest, error);
+}
+
+void Mle::HandleDataRequest(const Message &aMessage, const Ip6::MessageInfo &aMessageInfo, const Neighbor *aNeighbor)
+{
+    otError error = OT_ERROR_NONE;
+
+    OT_UNUSED_VARIABLE(aMessage);
+    OT_UNUSED_VARIABLE(aNeighbor);
+
+    Log(kMessageReceive, kTypeDataRequest, aMessageInfo.GetPeerAddr());
+
+#if OPENTHREAD_FTD
+    if (aNeighbor && aNeighbor->IsStateValid())
+    {
+        error = Get<MleRouter>().HandleDataRequest(aMessage, aMessageInfo);
+    }
+    else
+#endif
+    {
 #if OPENTHREAD_CONFIG_MAC_SSED_TO_SSED_LINK_ENABLE
+        error = HandleDataRequest(aMessage, aMessageInfo);
+#endif
+    }
+
+    LogProcessError(kTypeDataRequest, error);
+}
+
+#if OPENTHREAD_CONFIG_MAC_SSED_TO_SSED_LINK_ENABLE
+otError Mle::HandleLinkRequest(const Message &aMessage, const Ip6::MessageInfo &aMessageInfo)
+{
+    otError         error    = OT_ERROR_NONE;
+    Child *         peerSsed = nullptr;
+    Challenge       challenge;
+    uint16_t        version;
+    uint16_t        sourceAddress;
+    uint8_t         mode;
+    Mac::ExtAddress extAddr;
+
+    // Challenge
+    SuccessOrExit(error = ReadChallenge(aMessage, challenge));
+
+    // Version
+    SuccessOrExit(error = Tlv::Find<VersionTlv>(aMessage, version));
+    VerifyOrExit(version >= OT_THREAD_VERSION_1_2, error = OT_ERROR_DROP);
+
+    // Mode
+    SuccessOrExit(error = Tlv::Find<ModeTlv>(aMessage, mode));
+    VerifyOrExit(!(mode & (DeviceMode::kModeFullThreadDevice | DeviceMode::kModeRxOnWhenIdle)), error = OT_ERROR_DROP);
+
+    // Source Address
+    SuccessOrExit(error = Tlv::Find<SourceAddressTlv>(aMessage, sourceAddress));
+
+    if (IsActiveRouter(sourceAddress))
+    {
+        // Ignore router's Link Request
+        ExitNow(error = OT_ERROR_DROP);
+    }
+
+    // Find Child
+    aMessageInfo.GetPeerAddr().GetIid().ConvertToExtAddress(extAddr);
+
+    peerSsed = Get<ChildTable>().FindChild(extAddr, Child::kInStateAnyExceptInvalid);
+
+    if (peerSsed == nullptr)
+    {
+        VerifyOrExit((peerSsed = Get<ChildTable>().GetNewChild()) != nullptr, error = OT_ERROR_NO_BUFS);
+
+        // MAC Address
+        peerSsed->SetExtAddress(extAddr);
+        peerSsed->GetLinkInfo().Clear();
+        peerSsed->GetLinkInfo().AddRss(aMessageInfo.GetThreadLinkInfo()->GetRss());
+        peerSsed->ResetLinkFailures();
+        peerSsed->SetState(Neighbor::kStateLinkRequest);
+        peerSsed->SetDeviceMode(DeviceMode(DeviceMode::kModeRxOnWhenIdle));
+        peerSsed->SetRloc16(sourceAddress);
+    }
+
+    // Send Response
+    SuccessOrExit(error = SendLinkAccept(aMessageInfo, peerSsed, challenge, true));
```

Comment:
As mentioned above, I think we can use `SendLinkAcceptAndRequest` here for better readibility.